includes/class-newspack-popups.php

```diff
@@ -941,23 +941,90 @@ public static function archive_campaign( $id, $status ) {
 		update_term_meta( $id, self::NEWSPACK_POPUPS_TAXONOMY_STATUS, $status ? 'archive' : '' );
 	}
 
+	/**
+	 * Remove the default category from the given post, if it's the only category applied to that post.
+	 *
+	 * @param int $post_id ID of the post.
+	 */
+	private static function remove_default_category( $post_id ) {
+		$default_category_id = (int) get_option( 'default_category', false );
+		$post_categories     = wp_get_post_categories( $post_id );
+		if ( 1 === count( $post_categories ) && reset( $post_categories ) === $default_category_id ) {
+			wp_remove_object_terms( $post_id, $default_category_id, 'category' );
+		}
+	}
+
 	/**
 	 * Prevent setting the default category when publishing.
 	 *
 	 * @param string $new_status New status.
 	 * @param string $old_status Old status.
 	 * @param bool   $post Post.
 	 */
-	public static function remove_default_category( $new_status, $old_status, $post ) {
+	public static function prevent_default_category_on_publish( $new_status, $old_status, $post ) {
 		if ( self::NEWSPACK_POPUPS_CPT === $post->post_type && 'publish' !== $old_status && 'publish' === $new_status ) {
-			$default_category_id = (int) get_option( 'default_category', false );
-			$popup_has_category  = has_category( $default_category_id, $post->ID );
-			if ( $popup_has_category ) {
-				wp_remove_object_terms( $post->ID, $default_category_id, 'category' );
-			}
+			self::remove_default_category( $post->ID );
 		}
 	}
 
+	/**
+	 * When a category is deleted, any posts that have only that category assigned
+	 * are automatically assigned the site's default category (usually "Uncategorized").
+	 * We want to prevent this behavior for prompts, as prompts with the default
+	 * category will only appear on posts with that category.
+	 *
+	 * @param int    $deleted_term ID of the term being deleted.
+	 * @param string $taxonomy Name of the taxonomy the term belongs to.
+	 *
+	 * @return int The number of prompts affected by this callback.
+	 */
+	public static function prevent_default_category_on_term_delete( $deleted_term, $taxonomy ) {
+		// We only care about categories.
+		if ( 'category' !== $taxonomy ) {
+			return false;
+		}
+
+		$default_category_id = (int) get_option( 'default_category', false );
+		if ( false === $default_category_id ) {
```

Comment:
This variable is being casted to `int` so it'll never be `false` for this check. Maybe `empty()` is a safe assumption?

Reply:
Good call, updated in 692ac31.